workflows/cloudify_system_workflows/snapshot.py

```diff
@@ -0,0 +1,574 @@
+                      ' "q=select * from /.*/" > {0}')
+_INFLUXDB_RESTORE_CMD = ('cat {0} | while read -r line; do curl -X POST '
+                         '-d "[${{line}}]" "http://localhost:8086/db/cloudify/'
+                         'series?u=root&p=root" ;done')
+_STORAGE_INDEX_NAME = 'cloudify_storage'
+_EVENTS_INDEX_NAME = 'cloudify_events'
+
+
+class _DictToAttributes(object):
+    def __init__(self, dic):
+        self._dict = dic
+
+    def __getattr__(self, name):
+        return self._dict[name]
+
+
+def _get_json_objects(f):
+    def chunks(g):
+        ch = g.read(10000)
+        yield ch
+        while ch:
+            ch = g.read(10000)
+            yield ch
+
+    s = ''
+    n = 0
+    decoder = json.JSONDecoder()
+    for ch in chunks(f):
+        s += ch
+        try:
+            while s:
+                obj, idx = decoder.raw_decode(s)
+                n += 1
+                yield json.dumps(obj)
+                s = s[idx:]
+        except:
+            pass
+
+    assert not n or not s
+
+
+def _copy_data(archive_root, config, to_archive=True):
+    DATA_TO_COPY = [
+        (config.file_server_blueprints_folder, 'blueprints'),
+        (config.file_server_uploaded_blueprints_folder, 'uploaded-blueprints')
+    ]
+
+    # files with constant relative/absolute paths
+    for (p1, p2) in DATA_TO_COPY:
+        if p1[0] != '/':
+            p1 = os.path.join(config.file_server_root, p1)
+        if p2[0] != '/':
+            p2 = os.path.join(archive_root, p2)
+        if not to_archive:
+            p1, p2 = p2, p1
+
+        if not os.path.exists(p1):
+            continue
+
+        if os.path.isfile(p1):
+            shutil.copy(p1, p2)
+        else:
+            if os.path.exists(p2):
+                shutil.rmtree(p2, ignore_errors=True)
+            shutil.copytree(p1, p2)
+
+
+def _create_es_client(config):
+    return elasticsearch.Elasticsearch(hosts=[{'host': config.db_address,
+                                               'port': int(config.db_port)}])
+
+
+def _except_types(s, *args):
+    return (e for e in s if e['_type'] not in args)
+
+
+def _clean_up_db_before_restore(es_client, wf_exec_id):
+    s = elasticsearch.helpers.scan(es_client)
+    for doc in _except_types(s, 'provider_context', 'snapshot'):
+        if doc['_id'] != wf_exec_id:
+            doc['_op_type'] = 'delete'
+            yield doc
+
+
+def _create(ctx, snapshot_id, config, include_metrics, include_credentials,
+            **kw):
+    tempdir = tempfile.mkdtemp('-snapshot-data')
+
+    snapshots_dir = os.path.join(
+        config.file_server_root,
+        config.file_server_snapshots_folder
+    )
+
+    if not os.path.exists(snapshots_dir):
+        os.makedirs(snapshots_dir)
+
+    metadata = {}
+
+    # files/dirs copy
+    _copy_data(tempdir, config)
+
+    # elasticsearch
+    es = _create_es_client(config)
+    has_cloudify_events = es.indices.exists(index=_EVENTS_INDEX_NAME)
```

Comment:
We changed logstash configuration to create a new events index per day/week and the index name is: "logstash-*".
Look here: https://github.com/cloudify-cosmo/cloudify-manager/blob/master/rest-service/manager_rest/resources.py#L1183